sources/dogtag/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+/*!
+dogtag is a set of tools that detect the hostname of a bottlerocket server/instance and prints it to stdout.
+if the tool is called in an environment it cannot resolve the hostname it will error out.
+
+Currently the following hostname tools are implemented:
+
+* 01-imds - Fetches hostname from the Instance Metadata via IMDS
+* 00-reverse-dns - Uses reverse dns lookup to resolve the hostname
+ */
+use argh::FromArgs;
+use log::debug;
+use snafu::ResultExt;
+use std::{fs, process};
+
+const DOGTAG_BIN_PATH: &str = "/var/bottlerocket/dogtag";
+
+/// CLi defines the standard cmdline interface for all hostname handlers
+#[derive(FromArgs)]
+#[argh(description = "hostname resolution tool")]
+pub struct Cli {
+    #[argh(option)]
+    #[argh(description = "ip_address of the host")]
+    pub ip_address: String,
+}
+
+pub type Result<T> = std::result::Result<T, error::Error>;
+
+/// find_hostname will utilize the helpers located in /var/bottlerocket/dogtag/ to try and discover the hostname
+pub async fn find_hostname() -> Result<String> {
+    debug!(
+        "attempting to discover hostname helpers in {}",
+        DOGTAG_BIN_PATH
+    );
+    let walker = fs::read_dir(DOGTAG_BIN_PATH).context(error::IoSnafu)?;
+    let mut hostname_helpers = Vec::new();
+    for entry in walker.flatten() {
+        let path = entry.path();
+        if path.is_file() {
+            hostname_helpers.push(path.clone());
+        }
+    }
+    // We want to do reverse sort as we want to prioritize higher numbers first
```

Comment:
should we allow for a usecase where an OOTBuilder would want to add a dogtag resolver at a lower priority than the default ones? Like perhaps put the two defaults on `10-` and `20-`? with a 2-digit prefix to the binaries, there should be plenty of wiggle room -- hard to imagine a distro with >99 ways to resolve a hostname.

Reply:
Yeah i think thats a good idea, i'll bump them to 10- and 20-